eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -0,0 +1,183 @@
+import org.opencb.biodata.models.variant.VariantSourceEntry;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Created by pagarcia on 27/06/2016.
+ */
+public class BiodataVariantToVariantContextConverter {
+
+    public static final String GENOTYPE_KEY = "GT";
+    private final VariantContextBuilder variantContextBuilder;
+    private final List<VariantSource> sources;
+    private CellbaseWSClient cellbaseClient;
+    private String regionSequence;
+    private Map<String, Map<String,String>> studiesSampleNamesEquivalences;
+    private static final int NO_CALL_ALLELE_INDEX = 2;
+
+    public BiodataVariantToVariantContextConverter(List<VariantSource> sources, CellbaseWSClient cellbaseWSClient,
+                                                   Map<String, Map<String,String>> studiesSampleNamesEquivalences)
+    {
+        this.sources = sources;
+        this.cellbaseClient = cellbaseWSClient;
+        this.studiesSampleNamesEquivalences = studiesSampleNamesEquivalences;
+        variantContextBuilder = new VariantContextBuilder();
+    }
+
+    public VariantContext transform(Variant variant, Region region) throws CellbaseSequenceDownloadError {
+        String[] allelesArray = getAllelesArray(variant, region);
+
+        Set<Genotype> genotypes = getGenotypes(variant, allelesArray);
+
+        VariantContext variantContext = variantContextBuilder
+                .chr(variant.getChromosome())
+                // TODO: check start and end for indels
+                .start(variant.getStart())
+                .stop(variant.getEnd())
+                .noID()
+                .alleles(allelesArray)
+                .unfiltered()
+                .genotypes(genotypes).make();
+        return variantContext;
+    }
+
+    private String[] getAllelesArray(Variant variant, Region region) throws CellbaseSequenceDownloadError {
+        String[] allelesArray;
+        // if there are indels, we cannot use the normalized alleles, (hts forbids empty alleles) so we have to take them from cellbase
+        if (variant.getReference().isEmpty() || variant.getAlternate().isEmpty()) {
+            String contextNucleotide;
+            if (region != null) {
+                contextNucleotide = getContextNucleotideFromCellbaseCachingRegions(variant, variant.getStart(), region);
+            } else {
+                contextNucleotide = getContextNucleotideFromCellbase(variant, variant.getStart());
+            }
+
+            allelesArray = new String[] {contextNucleotide + variant.getReference(), contextNucleotide + variant.getAlternate()};
+            variant.setEnd(variant.getStart() + allelesArray[0].length() - 1);
+        } else {
+            allelesArray = new String[] {variant.getReference(), variant.getAlternate()};
+        }
+
+        return allelesArray;
+    }
+
+    private String getContextNucleotideFromCellbase(Variant variant, Integer start) throws CellbaseSequenceDownloadError {
```

Comment:
I suspect that for certain species that are not fully supported by Ensembl (right now parasites), using CellBase will be an issue. We will need to think of an alternative...